qemu/tests/qmp_command.py

```diff
@@ -19,167 +21,127 @@ def run(test, params, env):
+        :param key: expect result
+        :type key: str
+        :param val: expect result
+        :type val: str or None(if check_item_in_pair=False)
+        :param check_item_in_pair: If expect result is dict (True) or str (False)
+        :type check_item_in_pair: bool.
+
+        :return check result
+        :rtype: bool
+        """
+        for element in qmp_o:
+            if isinstance(element, dict):
+                if _check_dict(element, key, val, check_item_in_pair):
+                    return True
+            elif isinstance(element, list):
+                if check_list(element, key, val, check_item_in_pair):
+                    return True
+            elif element != '' and not check_item_in_pair:
+                if strict_match:
+                    if operator.eq(key, element):
+                        return True
+                else:
+                    if key in str(element):
+                        return True
+        return False
+
+    def _check_dict(dic, key, val=None, check_item_in_pair=True):
+        """
+        Check if the expect key, val are contained in QMP output dic.
+
+        :param dic: content of QMP command return value
+        :type dic: dict
+        :param key: expect result
+        :type key: str
+        :param val: expect result
+        :type val: str or None(if check_item_in_pair=False)
+        :param check_item_in_pair: If expect result is dict or str
+        :type check_item_in_pair: bool. Means expect result is dict or str.
+
+        :return check result
+        :rtype: bool
+        """
+        if key in dic and not check_item_in_pair:
+            return True
+        elif key in dic and val == dic[key]:
+            return True
+        else:
+            for value in dic.values():
+                if isinstance(value, dict):
+                    if _check_dict(value, key, val, check_item_in_pair):
+                        return True
+                elif isinstance(value, list):
+                    if check_list(value, key, val, check_item_in_pair):
+                        return True
+                elif value != '' and not check_item_in_pair:
+                    if strict_match:
+                        if operator.eq(key, value):
+                            return True
+                    else:
+                        if key in str(value):
+                            return True
+            return False
+
+    def check_result(qmp_o, expect_o=None, exception_list=""):
         """
-        Check test result with difference way accoriding to
-        result_check.
+        Check test result with difference way according to result_check.
         result_check = equal, will compare cmd_return_value with qmp
                        command output.
         result_check = contain, will try to find cmd_return_value in qmp
                        command output.
-        result_check = m_equal_q, will compare key value in monitor command
```

Comment:
Should we list all the possible values here? I see others like "post_not_contain", "post_contain", a bit confusing.

Reply:
Thanks,  will add doc str for `not_contain`,   no need to add `post_not_contain` and `post_contain` since it will be handled before this function be called.